rust_src/src/minibuf.rs

```diff
@@ -51,4 +51,18 @@ pub fn set_minibuffer_window(window: LispObject) -> LispObject {
     window
 }
 
+/// Return current depth of activations of minibuffer,
+/// a nonnegative integer.
+#[lisp_fn]
+pub fn minibuffer_depth() -> EmacsInt {
+    unsafe { minibuf_level }
+}
+
+/// Return the prompt string of the currently active
+/// minibuffer. If no minibuffer is active return nil.
+#[lisp_fn]
+pub fn minibuffer_prompt() -> LispObject {
+    LispObject::from_raw(unsafe { minibuf_prompt })
```

Comment:
This needs to call `copy_sequence` in the Rust too, as far as I can see.

Reply:
is `copy_sequence` already ported to Rust? if so, i'm having trouble grepping for it. otherwise i assume `Fcopy_sequence` should work alright.

Reply:
Correct on both counts. It is not ported and `Fcopy_sequence` is the right thing to use currently.

Reply:
Also, we removed the need for the call to `from_raw`. Just pass the `minibuf_promp` straight into `Fcopy_sequence`.